mscore/pa.cpp

```diff
@@ -274,6 +275,103 @@ void Portaudio::midiRead()
       }
 
 //---------------------------------------------------------
+//   putEvent
+//---------------------------------------------------------
+
+// Prevent killing sequencer with wrong data
+#define less128(__less) ((__less >=0 && __less <= 127) ? __less : 0)
+
+void Portaudio::putEvent(const NPlayEvent& e, unsigned framePos)
+      {
+      PortMidiDriver* portMidiDriver = static_cast<PortMidiDriver*>(midiDriver);
+      if (!portMidiDriver || !portMidiDriver->getOutputStream())
+            return;
+
+      unsigned timestamp = seq->getInitialPlayTimestampWithLatency() + (framePos / MScore::sampleRate) * 1000; // convert notes play time from frames to real milliseconds
```

Comment:
I just realized I should probably do (framePos * 1000) first before dividing by sample rate, because of the fact that the decimals will be lost in the integer division.